hal/armv7a/pmap.c

```diff
@@ -353,32 +353,35 @@ int pmap_enter(pmap_t *pmap, addr_t pa, void *va, int attr, page_t *alloc)
 }
 
 
-int pmap_remove(pmap_t *pmap, void *vaddr)
+int pmap_remove(pmap_t *pmap, void *vstart, void *vend)
 {
 	unsigned int pdi, pti;
 	addr_t addr;
-	unsigned char asid;
 	spinlock_ctx_t sc;
-
-	pdi = ID_PDIR((ptr_t)vaddr);
-	pti = ID_PTABLE((ptr_t)vaddr);
+	ptr_t vaddr;
 
 	hal_spinlockSet(&pmap_common.lock, &sc);
-	if (!(addr = pmap->pdir[pdi])) {
-		hal_spinlockClear(&pmap_common.lock, &sc);
-		return EOK;
-	}
 
-	/* Map page table corresponding to vaddr */
-	asid = pmap_common.asids[pmap->asid_ix];
-	_pmap_mapScratch(addr, asid);
+	const u8 asid = pmap_common.asids[pmap->asid_ix];
 
-	if (pmap_common.sptab[pti] == 0) {
-		hal_spinlockClear(&pmap_common.lock, &sc);
-		return EOK;
-	}
+	for (vaddr = (ptr_t)vstart; vaddr < (ptr_t)vend; vaddr += SIZE_PAGE) {
+		pdi = ID_PDIR(vaddr);
+		pti = ID_PTABLE(vaddr);
+
+		addr = pmap->pdir[pdi];
+		if (addr == PDIR_TYPE_INVALID) {
+			continue;
+		}
 
-	_pmap_writeEntry(pmap_common.sptab, vaddr, 0, 0, asid);
+		/* Map page table corresponding to vaddr */
+		_pmap_mapScratch(addr, asid);
```

Comment:
There's a good chance that we will map the same page after first iteration - would be nice to skip this when possible